packages/application/src/lab.ts

```diff
@@ -206,10 +206,19 @@ export class JupyterLab extends JupyterFrontEnd<ILabShell> {
    */
   protected evtKeydown(event: KeyboardEvent): void {
     // Process select keys which may call `preventDefault()` immediately
+    // TODO: generalise with https://github.com/jupyterlab/lumino/issues/688
     if (
+      // Navigation shortcuts which do not result in user input
+      // (except for Tab which is handled specially).
       ['Tab', 'ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(
         event.key
-      )
+      ) ||
+      // Saving shortcut which competes with the default browser action;
+      // metaKey is `Command` on Mac.
+      ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() == 's') ||
+      // Command + Shift + C on Mac should only open the Command Palette
+      // (rather than also opening the Dev Tools)
+      (event.metaKey && event.shiftKey && event.key.toLowerCase() == 'c')
```

Comment:
Does this condition also handle on non-mac devices?

Reply:
I guess this would also fix https://github.com/jupyterlab/jupyterlab/issues/15936

Reply:
Yes, though honestly I'm also considering just reverting this handler logic altogether until we get the the improved handling in lumino.

Reply:
Right, but I guess we'll still want this as a hot fix for now?

Reply:
No, I mean closing this one and merging revert PR https://github.com/jupyterlab/jupyterlab/pull/15938. This is mostly because I think we should not be breaking shortcuts contributed by extensions for which we do not have ability to provide a set of workaround like in this PR.

Reply:
Ah ok got it :+1: